subsys/bluetooth/host/att.c

```diff
@@ -539,52 +540,41 @@ static void bt_att_sent(struct bt_l2cap_chan *ch)
 	(void)process_queue(chan, &att->tx_queue);
 }
 
-static void chan_cfm_sent(struct bt_conn *conn, struct bt_att_tx_meta_data *user_data)
-{
-	struct bt_att_tx_meta_data *data = user_data;
-	struct bt_att_chan *chan = data->att_chan;
-
-	LOG_DBG("chan %p", chan);
-
-	tx_meta_data_free(data);
-}
-
-static void chan_rsp_sent(struct bt_conn *conn, struct bt_att_tx_meta_data *user_data)
-{
-	struct bt_att_tx_meta_data *data = user_data;
-	struct bt_att_chan *chan = data->att_chan;
-
-	LOG_DBG("chan %p", chan);
-
-	tx_meta_data_free(data);
-}
-
-static void chan_req_sent(struct bt_conn *conn, struct bt_att_tx_meta_data *user_data)
+static void chan_rebegin_att_timeout(struct bt_att_tx_meta_data *user_data)
 {
 	struct bt_att_tx_meta_data *data = user_data;
 	struct bt_att_chan *chan = data->att_chan;
 
 	LOG_DBG("chan %p chan->req %p", chan, chan->req);
 
-	/* Start timeout work */
+	if (!atomic_test_bit(chan->flags, ATT_CONNECTED)) {
+		LOG_ERR("ATT channel not connected");
+		return;
+	}
```

Comment:
Does this ensure that the buffer is properly destroyed?

Reply:
yes. The buffer is destroyed prior to calling the callback (this fn).